File: Factory/C#/Logistics/Factory/AirLogistics.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Factory
{
    public class AirLogistics : Logistics
    {
        public override Transport CreateTransport(string whatToDeliver, Coordinate from, Coordinate to)
        {
            var transportPoints = PlanTransport(from, to);
            return new Plane(whatToDeliver, transportPoints);
        }

        protected override Coordinate[] PlanTransport(Coordinate from, Coordinate to)
        {
            return new Coordinate[] { from, to };
        }
    }
}

File: Factory/Java/LogisticsFactory/src/LogisticsFactory/AirLogistics.java
package LogisticsFactory;

public class AirLogistics extends Logistics  {

	@Override
	protected Transport createTransport(String whatToDeliver, Coordinate from, Coordinate to) {
		var transportPoints = planTransport(from, to);
		return new Plane(whatToDeliver, transportPoints);
	}

	@Override
	protected Coordinate[] planTransport(Coordinate from, Coordinate to) {

		return new Coordinate [] {from, to};
	}

}
